### ISSUE TITLE
In Parser.php line 154: Notice: Trying to access array offset on value of type int 

### ISSUE THREAD

**Issue by maxson007** [smalot/pdfparser#267]
Hi
I have an error :
`In Parser.php line 154: Notice: Trying to access array offset on value of type int 
`

**Comment by j0k3r** [comment]
Can you provide the PDF which generate that notice? So we can provide a fix.

**Comment by BumbleBeeast** [comment]
Hey Guys,

i have the error with the attached file. I hope this information helps.

These two lines repeat hundred of times:
Notice: Trying to access array offset on value of type int smalot\pdfparser\src\Smalot\PdfParser\Parser.php on line 217
Notice: Trying to access array offset on value of type int smalot\pdfparser\src\Smalot\PdfParser\Parser.php on line 152

After that it runs into a fatal error:
PHP Fatal error:  Uncaught Exception: Missing catalog. in smalot\pdfparser\src\Smalot\PdfParser\Document.php:240
Stack trace:
#0 smalot\pdfparser\src\Smalot\PdfParser\Document.php(251): Smalot\PdfParser\Document->getPages()
#1 commonfunctions.php(114): Smalot\PdfParser\Document->getText()
....
#5 {main}
  thrown in smalot\pdfparser\src\Smalot\PdfParser\Document.php on line 240



[route_melbourne.pdf](https://github.com/smalot/pdfparser/files/4797463/route_melbourne.pdf)

**Comment by k00ni** [comment]
I created a branch named `issue/267`, which contains a test case that should demonstrate the errors @BumbleBeeast and @maxson007 mentioned. It uses the PDF `route_melbourne.pdf` provided by @BumbleBeeast.

https://github.com/smalot/pdfparser/blob/issue/267/tests/Integration/ParserTest.php#L79-L101

But i only could reproduce the `Missing Catalog.` exception.

These notifications seem to only appear in PHP 7.4+ i assume, based on https://travis-ci.org/github/smalot/pdfparser/jobs/699700902#L336.

**Comment by k00ni** [comment]
### Status update:

I merged #310, which fixes these notifications:

```
Notice: Trying to access array offset on value of type int src\Smalot\PdfParser\Parser.php on line 217
Notice: Trying to access array offset on value of type int src\Smalot\PdfParser\Parser.php on line 152
...
```

* [x] Fix `array offset on value of type int` notices 
* [ ] Exception `Missing Catalog.` is thrown for PDF https://github.com/smalot/pdfparser/files/4797463/route_melbourne.pdf
